### PR TITLE
[WIP]Initial commit - dynaconf integration for cfme_data

### PR DESCRIPTION
**Pull request by kedark3**
## Purpose or Intent

- __Using Dynaconf for Cfme_data.yaml__ This is initial draft. 

**requirement**  pip3 install "dynaconf[yaml]" 

**Usage:**
```
In [1]: from cfme.utils.config_data import cfme_data                                                                   

In [2]: cfme_data.vm_console                                                                                           
Out[2]: <Box: {'cert': {'country': 'US', 'state': 'North Carolina', 'city': 'Durham', 'organization': 'CFME', 'organizational_unit': 'QE', 'install_dir': '/var/www/miq/vmdb/certs'}, 'webmks_console': {'webmks_sdk_download_url': 'http://file.rdu.redhat.com/~kkulkarn/WebMKS_SDK_2.1.0.zip', 'webmks_sdk_extract_location': '/var/www/miq/vmdb/public/webmks'}}>

In [3]: cfme_data.rhv43.tags                                                                                           
Out[3]: <BoxList: ['rhevm', 'rhv4', 'v2v_group', 'complete', 'default', 'onprem', 'ds_stable']>

In [4]: cfme_data.db_backups.ec2_5540.version                                                                          
Out[4]: '5.5.4.0'

```

Just need to add env-name as `cfme-qe-test-envs:` at the top of cfme_data.yaml and add one indent everything else.

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/utils/config_data.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from dynaconf import LazySettings
+# from yaycl import AttrDict
```
Comment (jawatts) [inline cfme/utils/config_data.py]: You can remove this

**Comment by kedark3** [comment]
@mshriver @jawatts I think Travis is failing because of conf/cfme_data.yaml is not yet updated?

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 5c3f8928409c6a7d24ffdae27ef0cded1a972ee1

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`candu_db_restore`** is used in the following files:
- **cfme/tests/candu/test_graph_groupbytag.py**
    - *test_tagwise*

The *global* fixture **`cfme_data`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_basic.py**
- **cfme/tests/infrastructure/test_host_provisioning.py**
    - *test_host_provisioning*

The *local* fixture **`metrics_collection`** is used in the following files:
- **cfme/tests/candu/test_utilization_metrics.py**
    - *test_raw_metric_vm_cpu*
    - *test_raw_metric_vm_memory*
    - *test_raw_metric_vm_network*
    - *test_raw_metric_vm_disk*
    - *test_raw_metric_host_cpu*
    - *test_raw_metric_host_memory*
    - *test_raw_metric_host_network*
    - *test_raw_metric_host_disk*

The *local* fixture **`cfme_vhd`** is used in the following files:
- **cfme/tests/infrastructure/test_scvmm_specific.py**
    - *scvmm_appliance*

The *local* fixture **`local_setup_provider`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_vm_instance_analysis.py**
    - *ssa_single_vm*
    - *test_ssa_template*
    - *test_ssa_compliance*

The *local* fixture **`compliance_vm`** is used in the following files:
- **cfme/tests/control/test_compliance.py**
    - *test_check_package_presence*
    - *test_check_files*

The *local* fixture **`configure_vddk`** is used in the following files:
- **cfme/tests/infrastructure/test_ssa_vddk.py**
    - *test_ssa_vddk*

The *local* fixture **`db_restore`** is used in the following files:
- **cfme/tests/optimize/test_bottlenecks.py**
    - *test_bottlenecks_report_event_groups*
    - *test_bottlenecks_report_show_host_events*
    - *test_bottlenecks_report_time_zone*
    - *test_bottlenecks_summary_event_groups*
    - *test_bottlenecks_summary_show_host_events*
    - *test_bottlenecks_summary_time_zone*

The *local* fixture **`template_folder`** is used in the following files:
- **cfme/tests/pod/test_appliance_crud.py**
    - *new_ssa_image*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit d6b262fb453b59b329169064574c1f19e0a599bb

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`candu_db_restore`** is used in the following files:
- **cfme/tests/candu/test_graph_groupbytag.py**
    - *test_tagwise*

The *global* fixture **`cfme_data`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_basic.py**
- **cfme/tests/infrastructure/test_host_provisioning.py**
    - *test_host_provisioning*

The *local* fixture **`metrics_collection`** is used in the following files:
- **cfme/tests/candu/test_utilization_metrics.py**
    - *test_raw_metric_vm_cpu*
    - *test_raw_metric_vm_memory*
    - *test_raw_metric_vm_network*
    - *test_raw_metric_vm_disk*
    - *test_raw_metric_host_cpu*
    - *test_raw_metric_host_memory*
    - *test_raw_metric_host_network*
    - *test_raw_metric_host_disk*

The *local* fixture **`cfme_vhd`** is used in the following files:
- **cfme/tests/infrastructure/test_scvmm_specific.py**
    - *scvmm_appliance*

The *local* fixture **`local_setup_provider`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_vm_instance_analysis.py**
    - *ssa_single_vm*
    - *test_ssa_template*
    - *test_ssa_compliance*

The *local* fixture **`compliance_vm`** is used in the following files:
- **cfme/tests/control/test_compliance.py**
    - *test_check_package_presence*
    - *test_check_files*

The *local* fixture **`configure_vddk`** is used in the following files:
- **cfme/tests/infrastructure/test_ssa_vddk.py**
    - *test_ssa_vddk*

The *local* fixture **`db_restore`** is used in the following files:
- **cfme/tests/optimize/test_bottlenecks.py**
    - *test_bottlenecks_report_event_groups*
    - *test_bottlenecks_report_show_host_events*
    - *test_bottlenecks_report_time_zone*
    - *test_bottlenecks_summary_event_groups*
    - *test_bottlenecks_summary_show_host_events*
    - *test_bottlenecks_summary_time_zone*

The *local* fixture **`template_folder`** is used in the following files:
- **cfme/tests/pod/test_appliance_crud.py**
    - *new_ssa_image*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 1bc9788681d4a40026cdd493b6f02dc6b121b780

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`candu_db_restore`** is used in the following files:
- **cfme/tests/candu/test_graph_groupbytag.py**
    - *test_tagwise*

The *global* fixture **`cfme_data`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_basic.py**
- **cfme/tests/infrastructure/test_host_provisioning.py**
    - *test_host_provisioning*

The *local* fixture **`metrics_collection`** is used in the following files:
- **cfme/tests/candu/test_utilization_metrics.py**
    - *test_raw_metric_vm_cpu*
    - *test_raw_metric_vm_memory*
    - *test_raw_metric_vm_network*
    - *test_raw_metric_vm_disk*
    - *test_raw_metric_host_cpu*
    - *test_raw_metric_host_memory*
    - *test_raw_metric_host_network*
    - *test_raw_metric_host_disk*

The *local* fixture **`cfme_vhd`** is used in the following files:
- **cfme/tests/infrastructure/test_scvmm_specific.py**
    - *scvmm_appliance*

The *local* fixture **`local_setup_provider`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_vm_instance_analysis.py**
    - *ssa_single_vm*
    - *test_ssa_template*
    - *test_ssa_compliance*

The *local* fixture **`compliance_vm`** is used in the following files:
- **cfme/tests/control/test_compliance.py**
    - *test_check_package_presence*
    - *test_check_files*

The *local* fixture **`configure_vddk`** is used in the following files:
- **cfme/tests/infrastructure/test_ssa_vddk.py**
    - *test_ssa_vddk*

The *local* fixture **`db_restore`** is used in the following files:
- **cfme/tests/optimize/test_bottlenecks.py**
    - *test_bottlenecks_report_event_groups*
    - *test_bottlenecks_report_show_host_events*
    - *test_bottlenecks_report_time_zone*
    - *test_bottlenecks_summary_event_groups*
    - *test_bottlenecks_summary_show_host_events*
    - *test_bottlenecks_summary_time_zone*

The *local* fixture **`template_folder`** is used in the following files:
- **cfme/tests/pod/test_appliance_crud.py**
    - *new_ssa_image*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 63879e7ee8a9ce928415c4dd7b425625e0385e04

<details>
<summary>
<b>Show fixtures</b>
</summary>


The *global* fixture **`candu_db_restore`** is used in the following files:
- **cfme/tests/candu/test_graph_groupbytag.py**
    - *test_tagwise*

The *global* fixture **`cfme_data`** is used in the following files:
- **cfme/tests/ansible/test_embedded_ansible_basic.py**
- **cfme/tests/infrastructure/test_host_provisioning.py**
    - *test_host_provisioning*

The *local* fixture **`metrics_collection`** is used in the following files:
- **cfme/tests/candu/test_utilization_metrics.py**
    - *test_raw_metric_vm_cpu*
    - *test_raw_metric_vm_memory*
    - *test_raw_metric_vm_network*
    - *test_raw_metric_vm_disk*
    - *test_raw_metric_host_cpu*
    - *test_raw_metric_host_memory*
    - *test_raw_metric_host_network*
    - *test_raw_metric_host_disk*

The *local* fixture **`cfme_vhd`** is used in the following files:
- **cfme/tests/infrastructure/test_scvmm_specific.py**
    - *scvmm_appliance*

The *local* fixture **`local_setup_provider`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_vm_instance_analysis.py**
    - *ssa_single_vm*
    - *test_ssa_template*
    - *test_ssa_compliance*

The *local* fixture **`compliance_vm`** is used in the following files:
- **cfme/tests/control/test_compliance.py**
    - *test_check_package_presence*
    - *test_check_files*

The *local* fixture **`configure_vddk`** is used in the following files:
- **cfme/tests/infrastructure/test_ssa_vddk.py**
    - *test_ssa_vddk*

The *local* fixture **`db_restore`** is used in the following files:
- **cfme/tests/optimize/test_bottlenecks.py**
    - *test_bottlenecks_report_event_groups*
    - *test_bottlenecks_report_show_host_events*
    - *test_bottlenecks_report_time_zone*
    - *test_bottlenecks_summary_event_groups*
    - *test_bottlenecks_summary_show_host_events*
    - *test_bottlenecks_summary_time_zone*

The *local* fixture **`template_folder`** is used in the following files:
- **cfme/tests/pod/test_appliance_crud.py**
    - *new_ssa_image*

</details>

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by kedark3** [comment]
Closing this PR in lieu of #9376

**Pull request closed without merging**